tideways/provision.sh

```diff
@@ -0,0 +1,112 @@
+    sudo apt update
+    apt-get -y install mongodb-org re2c
+    sudo pecl install mongodb
+    ln -s /usr/lib/php/20151012/mongodb.so /usr/lib/php/20170718/mongodb.so
+    ln -s /usr/lib/php/20160303/mongodb.so /usr/lib/php/20170718/mongodb.so
+    phpenmod mongodb
+    # auto-remove records older than 2592000 seconds (30 days)
+    mongo xhprof --eval 'db.collection.ensureIndex( { "meta.request_ts" : 1 }, { expireAfterSeconds : 2592000 } )'
+    # indexes
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'meta.SERVER.REQUEST_TIME' : -1 } )"
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'profile.main().wt' : -1 } )"
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'profile.main().mu' : -1 } )"
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'profile.main().cpu' : -1 } )"
+    mongo xhprof --eval  "db.collection.ensureIndex( { 'meta.url' : 1 } )"
+    update-rc.d mongodb defaults
+    update-rc.d mongodb enable
+}
+
+echo "Installing Tideways & XHgui"
+if [[ ! -d "/srv/www/default/xhgui" ]]; then
+    if [[ -d "/etc/php/5.6/" ]]; then
+        echo "File copied for php 5.6"
+        cp "${DIR}/tideways.ini" "/etc/php/5.6/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/mongodb.ini" "/etc/php/5.6/mods-available/mongodb.ini"
+    fi
+    if [[ -d "/etc/php/7.0/" ]]; then
+        echo "File copied for php 7.0"
+        cp "${DIR}/tideways.ini" "/etc/php/7.0/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/mongodb.ini" "/etc/php/7.0/mods-available/mongodb.ini"
+        cp "${DIR}/xhgui-php.ini" "/etc/php/7.0/mods-available/xhgui.ini"
+    fi
+    if [[ -d "/etc/php/7.1/" ]]; then
+        echo "File copied for php 7.1"
+        cp "${DIR}/tideways.ini" "/etc/php/7.1/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/mongodb.ini" "/etc/php/7.1/mods-available/mongodb.ini"
+        cp "${DIR}/xhgui-php.ini" "/etc/php/7.1/mods-available/xhgui.ini"
+    fi
+    if [[ -d "/etc/php/7.2/" ]]; then
+        echo "File copied for php 7.2"
+        cp "${DIR}/tideways.ini" "/etc/php/7.2/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/xhgui-php.ini" "/etc/php/7.2/mods-available/xhgui.ini"
+        # For the default php version
+        cp "${DIR}/mongodb.ini" "/etc/php/7.2/mods-available/mongodb.ini"
+    fi
+    install_mongodb
+    install_tideways
+    phpenmod tideways_xhprof
+    echo -e "\nDownloading xhgui, see https://github.com/perftools/xhgui"
+    git clone "https://github.com/perftools/xhgui" "/srv/www/default/xhgui"
+    cd /srv/www/default/xhgui
+    php install.php
+    cp "${DIR}/config.php" "/srv/www/default/xhgui/config/config.php"
+    cp "${DIR}/tideways-header.php" "/srv/www/default/xhgui/config/tideways-header.php"
+    cp "${DIR}/nginx.conf" "/etc/nginx/custom-sites/xhgui.conf"
+    restart_php
+    service mongodb restart
+    php7.0 --ri tideways_xhprof
+    php7.1 --ri tideways_xhprof
```

Comment:
There's no command for 7.2, only 7.1 and 7.0, and no check to make sure that 7.0 or 7.1 are actually installed